src/components/Modals/index.tsx

```diff
@@ -18,76 +18,61 @@ interface BlockModalProps extends TransactionModalProps {
   disabledReason: string,
 }
 
-const tempParentDiv: CSSProperties = {
-  display: 'flex',
-  flexFlow: 'column nowrap',
-  alignItems: 'center', justifyContent: 'center',
-  width: '100%',
-}
-
-const tempButtonDiv: CSSProperties = {
-  display: 'flex',
-  flexFlow: 'row nowrap',
-  justifyContent: 'center', alignItems: 'center',
-  padding: 10,
-  width: '75%',
-}
-
-const tempDivStyle: CSSProperties = {
-  position: 'fixed', 
-  display: 'flex', flexFlow: 'column', justifyContent: 'center', alignItems: 'center',
-
-  color: 'white',
-  
-  height: '100vh', width: '100vw',
-  textAlign: 'center',
-}
-
-export const TransactionModal: React.SFC<TransactionModalProps> = ({ 
+export const TransactionModal: React.SFC<TransactionModalProps> = ({
   modalProps: {
-    header, 
+    header,
     body,
+    txData,
     button,
     error,
-  }, 
+  },
   activeProvider,
   closeModal,
 }) =>
-  <div style={tempDivStyle}>
-    <h1 className="modalH1">{header || 'Approving Tokens and Transferring'}</h1>
-    <p className="modalP">
+  <div className="modalDivStyle">
+    <h1>{header || 'Approving Tokens and Transferring'}</h1>
+    <p className="modalHeaderDescriptor">
       { body || `Please check your ${activeProvider || 'Provider'} notifications in extensions bar of your browser.` }
     </p>
-    {error && 
+    {txData && <div className="modalTXDataDiv">
+      <ul>
+        <li>{`Sell Amount:    ${txData.sellAmount}`}</li>
+        {(txData.tokenA.symbol || txData.tokenA.symbol) &&
```

Comment:
This should be `txData.tokenA.symbol || txData.tokenA.name` right?
What about address when there is no `symbol or name`?

Reply:
`address` is below anyways - figured dont render the component at all if there is no `name` or `symbol`. but good catch on the other!

Reply:
done